### PR TITLE
Nonlinear duration

### PR DESCRIPTION
**Pull request by Kaleab470**

### CODE REVIEW

File: examples/player/simid_player.html [hunk 1]
```diff
@@ -37,14 +37,17 @@
       let linearButton = document.getElementById("linear_ad");
       let creativeDimensions = document.getElementById("creative_dimensions");
       let mediaFile = document.getElementById("media_file");
+      let creativeduration = document.getElementById("creative_duration");
```
Comment (juliareichel) [inline examples/player/simid_player.html:40]: camelCase

File: examples/player/simid_player.js [hunk 2]
```diff
@@ -306,6 +306,12 @@ class SimidPlayer {
       this.simidIframe_.style.position = "absolute";
 
       this.contentVideoElement_.play();
+
+      const nonlinearDuration = document.getElementById('duration').value * 1000;
```
Comment (juliareichel) [inline examples/player/simid_player.js:310]: * nonLinearDuration

File: examples/player/simid_player.js [hunk 3]
```diff
@@ -430,6 +436,7 @@ class SimidPlayer {
 
     const creativeData = {
       'adParameters' : document.getElementById('ad_params').value,
+      'duration' : document.getElementById('duration').value,
```
Comment (damargulis) [inline examples/player/simid_player.js:439]: I think you only want to add this for nonlinear ads?

File: examples/player/simid_player.js [hunk 4]
```diff
@@ -306,6 +306,12 @@ class SimidPlayer {
       this.simidIframe_.style.position = "absolute";
 
       this.contentVideoElement_.play();
+
+      const nonlinearDuration = document.getElementById('duration').value * 1000;
+      setTimeout(() => {
```
Comment (ryanthompson591) [inline examples/player/simid_player.js:311]: no let's not use setTimeout.  It's possible the user pauses playback (like when they go to a different tab or whatever).  Instead let's use currentTime.

So add some code to the currentTime callback that just does a comparison.

If (this.nonlinearStarTime_ - this.vidoeElement_.currentTime > duration) {
  this.stopAd(StopCode.UNSPECIFIED);
}


Side note: I think we need another stop code. NON_LINEAR_DURATION_COMPLETE

https://github.com/InteractiveAdvertisingBureau/SIMID/issues/389

File: examples/player/simid_player.html (hunk 1 above)
Comment (Kaleab470) [reply examples/player/simid_player.html:40]: done

File: examples/player/simid_player.js (hunk 3 above)
Comment (Kaleab470) [reply examples/player/simid_player.js:439]: good catch

File: examples/player/simid_player.js (hunk 4 above)
Comment (Kaleab470) [reply examples/player/simid_player.js]: Done!

File: examples/player/simid_player.js [hunk 5]
```diff
@@ -237,10 +250,15 @@ class SimidPlayer {
...
+    for(let [key, func] of this.adVideoTrackingEvents_) {
       this.adVideoElement_.removeEventListener(key, func, true);
     }
-    this.videoTrackingEvents_.clear();
+
+    for(let [key, func] of this.contentVideoTrackingEvents_) {
+      this.contentVideoElement_.removeEventListener(key, func, true);
+    }
+
+    this.adVideoTrackingEvents_.clear();
```
Comment (ryanthompson591) [inline examples/player/simid_player.js:261]: add
this.contentVideoElement_.clear()

Comment (Kaleab470) [reply examples/player/simid_player.js:261]: Good catch!

**Pull request merged**